sharktank/tests/kernels/attention_template_test.py

```diff
@@ -0,0 +1,138 @@
+        L = 7  # Target Seq Len
+        S = 6  # Source Seq Len
+        Eqk = Ev = 64  # embedding dimensions with subscript identifiers
+
+        q = torch.rand([N, H, L, Eqk], dtype=dtype)
+        k = torch.rand([N, H, S, Eqk], dtype=dtype)
+        v = torch.rand([N, H, S, Ev], dtype=dtype)
+        # mask is same type as inputs, therefore its added to score
+        mask = None
+        scale = torch.tensor(1.0, dtype=dtype)
+        if use_mask:
+            mask = torch.rand([L, S], dtype=dtype)
+
+            res2 = kernels.masked_flash_attention(q, k, v, mask, scale=scale)
+
+        else:
+            res2 = kernels.flash_attention(q, k, v, scale)
+
+        ref = torch.nn.functional.scaled_dot_product_attention(
+            q, k, v, mask, scale=scale
+        )
+
+        torch.testing.assert_close(res2.to(dtype), ref, atol=atol, rtol=rtol)
+
+    @parameterized.expand(
+        [
+            # Todo: fixed unmasked.
+            # (torch.float32, False, False),
+            (torch.float32, False, True),
+            (torch.float16, True, True),
+            (torch.float8_e4m3fnuz, False, True),
+        ]
+    )
+    def test_export_dynamic(self, dtype, static, use_mask):
+        ops.attention_impls.register_attention_override_by_name(
+            "masked_flash_attention"
+        )
+        cast = False
+        if dtype == torch.float8_e4m3fnuz:
+            dtype = torch.float32
+            cast = True
+        H = 4  # Head dim
+        N = 3  # Batch Size
+        L = 19  # Target Seq Len
+        S = 19  # Source Seq Len
+        Eqk = Ev = 64  # embedding dimensions with subscript identifiers
+
+        q = torch.rand([N, H, L, Eqk], dtype=dtype)
+        k = torch.rand([N, H, S, Eqk], dtype=dtype)
+        v = torch.rand([N, H, S, Ev], dtype=dtype)
+        if use_mask:
+            # mask is same type as inputs, therefore its added to score
+            mask = torch.rand([L, S], dtype=dtype)
+        if cast:
+            q = q.to(torch.float8_e4m3fnuz)
+            k = q.to(torch.float8_e4m3fnuz)
+            v = v.to(torch.float8_e4m3fnuz)
+            if use_mask:
+                mask = mask.to(torch.float8_e4m3fnuz)
+        scale = torch.tensor(1.0, dtype=dtype)
+        dynamic_shapes = None
+        if not static:
+            L_dim = torch.export.Dim("L")
+            S_dim = torch.export.Dim("S")
+            dynamic_shapes = {
+                "q": {2: L_dim},
+                "k": {2: S_dim},
+                "v": {2: S_dim},
+                "mask": {},
+                "scale": {},
+            }
+            if use_mask:
+                dynamic_shapes["mask"] = {0: L_dim, 1: S_dim}
+
+        class MyModule(torch.nn.Module):
+            def forward(self, q, k, v, mask, scale):
+                return ops.scaled_dot_product_attention(
+                    q, k, v, a=mask, is_causal=False, scale=scale
+                )
+
+        mod = MyModule()
+        dtype = torch.dtype
+        if use_mask:
```

Comment:
You could simplify this without branching if `mask` is set to `None` when `not use_mask` is true.